xmtp_mls/src/identity.rs

```diff
@@ -533,16 +533,16 @@ impl Identity {
             return Ok(());
         }
 
-        self.rotate_key_package(provider, api_client).await?;
+        let kp_bytes = self.rotate_key_package(provider)?;
```

Comment:
Q: why we don't use the function from here:
https://github.com/xmtp/libxmtp/pull/1532/files?diff=split&w=1#diff-3c0189371525b6aae3be7e5a7b5b5682204ef341fb2744fc876148c5ecab8642R750

Since seems to has the same two lines we're having here. wdyt?

Reply:
The generic constraints are more lax here than where the rotate_and_upload_key_package function is defined on. Definitely a cleanup item for sure, but outside the scope of this PR I think, and this PR is already big enough :sweat_smile:

Reply:
Ended up rewriting this to provide better data integrity. Reuses the function now.